[BC] Allow only `bool` tensors as mask in `masked_select`

malfet opened the pull request:
`byte` support was marked as deprecated in 1.8, so it's fine to remove this in 2.1 (or even 2.0)
Deprecation warning was added by https://github.com/pytorch/pytorch/pull/22261

Also, fix bunch of syntactic errors in comments


cc @albanD

malfet committed:
[BC] Allow only `bool` tensors as mask in `masked_select`

`byte` support was marked as deprecated in 1.13, so it's fine to remove
it in 2.1

malfet commented:
@pytorchbot merge

Skylion007 commented on aten/src/ATen/native/TensorAdvancedIndexing.cpp:
```diff
@@ -1899,12 +1894,12 @@ static Tensor & masked_select_out_impl_cpu(Tensor & result, const Tensor & self,
   // TODO: Here can only use std::partial_sum for C++14,
-  // use std::exclusive_scan when PyTorch upgrades to C++17, which have better peformance.
+  // use std::exclusive_scan when PyTorch upgrades to C++17, which have better performance.
   // std::exclusive_scan(mask_long_data, mask_long_data + mask_long.numel(), mask_prefix_sum_data, 0);
```
@malfet can you swap out the comments to std::exclusive_scan now that we are in C++17?

malfet replied:
I'll do it (and few other changes) in followup PR (don't want to muddy this one with unrelated functional changes)

malfet replied:
https://github.com/pytorch/pytorch/pull/96217

malfet committed:
And fix tests

Do not test for `torch.uint8` and expect `torch.masked_select` to raise
an assert

malfet commented:
@pytorchbot merge

malfet committed:
And do the same for cuda

malfet committed:
And MPS

malfet commented:
@pytorchbot merge
